beacon/src/beacon.c

```diff
@@ -484,26 +515,47 @@ int beacon_loop()
 {
     _beacon_update_();
 
-    uint32_t lp_timer_status = 0, hp_timer_status = 0;
+    uint32_t lp_timer_status = 0, hp_timer_status = 0, alt_timer_status = 0;
 
     int err = 0;
     while (!err)
     {
-        // get most updated time
-        // Low-precision timer is synced, so accumulate status here
-        lp_timer_status += k_timer_status_get(&kernel_time_lp);
 
-        // update beacon clock using kernel. The addition is the number of
-        // periods elapsed in the internal timer
-        beacon_clock_increment(lp_timer_status);
+        alt_timer_status += k_timer_status_get(&kernel_time_alternater);
+        log_debugf("%d\r\n", alt_timer_status);
+        if (alt_timer_status % 6 == 0)
```

Comment:
@Disnaming I'm not a huge fan of this logic change. Can't we just add the alternate timer without modifying the existing ones?

Reply:
If I'm not mistaken, this is completely breaking the existing clocks by only updating them every 6 seconds, yeah?

Reply:
oop, it looks like I haven't pushed the changes we made on Monday. The way it was implemented then, the alt clock would tick every 10 seconds, and after 6 ticks it would call all of the original code that was already there (which I believe matches the periodicity of the existing clocks).

What is looks like on my end:
```
#ifdef BEACON_PLATFORM__ZEPHYR
int beacon_loop()
{
    _beacon_update_();

    uint32_t lp_timer_status = 0, hp_timer_status = 0, alt_timer_status = 0;

    int err = 0;
    while (!err)
    {

#ifdef BEACON_GAEN_ENABLED
        alt_timer_status += k_timer_status_get(&kernel_time_alternater);
        if (alt_timer_status % (num_ms_in_min / alt_time_in_ms) == 0)
        {
#endif
            // get most updated time
            // Low-precision timer is synced, so accumulate status here
            lp_timer_status += k_timer_status_get(&kernel_time_lp);

            // update beacon clock using kernel. The addition is the number of
            // periods elapsed in the internal timer
            beacon_clock_increment(lp_timer_status);

            // high-precision collects the raw number of expirations
            hp_timer_status = k_timer_status_get(&kernel_time_hp);
#ifdef BEACON_GAEN_ENABLED
        }

        _alternate_advertisement_content_(alt_timer_status);
        alt_timer_status += k_timer_status_sync(&kernel_time_alternater);

#else

        // // Wait for a clock update, this blocks until the internal timer
        // // period expires, indicating that at least one unit of relevant beacon
        // // time has elapsed. timer status is reset here
        lp_timer_status = k_timer_status_sync(&kernel_time_lp);

#endif
    }
    return err;
}
#endif
```